### PR TITLE
www: add video poster, preload, & lazy load iframes

### PR DESCRIPTION
**Pull request by neo773**
## 🎯 Changes

This PR adds poster attribute for preview while the video is loading & preload tag in header for performance

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: www/src/components/Preview.tsx [hunk 1]
```diff
@@ -1,6 +1,26 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 
 export const Preview = () => {
+  function getWindowDimensions() {
```
Comment (KATT) [inline www/src/components/Preview.tsx:4]: You need to do `if (typeof window === "undefined") { return something }` here

**Review by KATT — CHANGES_REQUESTED** [review]
Instead of doing this with a hook and calculating the viewport size, I think we should do it with CSS - having two `<figure>`s or whatever that are hidden/block based on CSS. That would get rid of the build failing without doing special hacks too

**Review by KATT — APPROVED** [review]

File: www/src/pages/index.tsx [hunk 2]
```diff
@@ -139,7 +139,6 @@ const HomeContent: React.FC = () => {
               })
             }
             frameBorder="0"
-            loading="lazy"
```
Comment (neo773) [inline www/src/pages/index.tsx]: We need this, makes a huge improvement in page load time and lighthouse score

File: www/src/pages/index.tsx [hunk 3]
```diff
@@ -245,6 +245,28 @@ const HomeHead: React.FC = () => {
...
+        href="https://assets.trpc.io/www/v10/v10-dark-landscape.mp4"
+        as="video"
+        type="video/mp4"
+      />
+      <link
+        rel="preload"
+        href="https://assets.trpc.io/www/v10/preview-dark.mp4"
+        as="video"
+        type="video/mp4"
+      />
```
Comment (KATT) [inline www/src/pages/index.tsx:259]:
```suggestion
```

Comment (KATT) [reply www/src/pages/index.tsx:259]: I think these are not really needed and will be a bit too "greedy"

File: www/src/pages/index.tsx (hunk 2 above)
Comment (KATT) [reply www/src/pages/index.tsx]: Ah i see

Comment (neo773) [reply www/src/pages/index.tsx]: Yeah, and sorry about keeping these PRs open haven't had a chance yet to dig back in.

Comment (KATT) [reply www/src/pages/index.tsx]: Thanks!

**Review by KATT — APPROVED** [review]

**Pull request merged**